### PR TITLE
8 profile management

### PR DESCRIPTION
**Pull request by Ismoil-JS**
get profile with auth token ==>
Screenshot 2024-10-23 at 14 14 44
get another user with another user's token ==>
Screenshot 2024-10-23 at 14 16 22
update user ==>
Screenshot 2024-10-23 at 14 24 30
update user with another user's token ==>
Screenshot 2024-10-23 at 14 25 09

### CODE REVIEW

File: src/modules/user/user.service.ts [hunk 1]
```diff
@@ -56,4 +56,41 @@ export class UserService {
...
+
+    return user
+  }
+
+  async updateProfile(id: string, payload: UserCreateDto): Promise<User> {
+    const user = await this.userRepository.findOne({
+      where: { id },
+    })
+
+    console.log(user, 'user')
```
Comment (mukhammadjon-ss) [inline src/modules/user/user.service.ts]: please remove these console.logs

File: src/modules/user/user.controller.ts [hunk 2]
```diff
@@ -95,3 +95,65 @@ userRouter.put(
     }
   },
 )
+
+userRouter.get(
+  '/:id',
+  TokenParserMiddleware,
+  IsUUID,
+  async (req: any, res: Response) => {
```
Comment (mukhammadjon-ss) [inline src/modules/user/user.controller.ts]: what is primary purpose of this controller ? Users can't see each other's profiles ?

**Review by mukhammadjon-ss — CHANGES_REQUESTED** [review]

File: src/modules/user/user.controller.ts (hunk 2 above)
Comment (Ismoil-JS) [reply src/modules/user/user.controller.ts]: made it visible for everyone

File: src/modules/user/user.service.ts (hunk 1 above)
Comment (Ismoil-JS) [reply src/modules/user/user.service.ts]: removed

**Pull request merged**